### ISSUE TITLE
Carousel examples obsolete and incomplete

### ISSUE THREAD

**Issue by mark-at-tusksoft** [chanan/BlazorStrap#378]
Similar vein as #246, some of these examples don't appear to work with the latest version.

On the "With Captions & Headers & Fade" example, the indicators element has the attribute  `ActiveIndexChangedEvent="@indexChanged"` where `indexChanged` is not defined.

Clickable image example code line 4 has an LShft operator `@for (int i = 0; i <<> items.Count; i++)` likely a typo?

Another useful example would include slides using static items with raw-html as that is a common use-case with most carousels, not just bootstrap's.

Additionally, a list of params and compatible BSCarousel* child components would be useful.

Aside: While I can appreciate the consistency with the original Bootstrap 4 carousel doc, the amount of vertical space the carousel consumes on a wide screen is rather obnoxious.

Lastly, I offer the following as a custom navigation example (see code below)

Components/BSCarouselCustomControls.razor
```@namespace MyProject.Components
@inherits BSCarouselCustomControlsBase

<DynamicElement @attributes="@UnknownParameters" TagName="div" class="@Classname" role="navigation">
    <div class="col-auto">
        @if (CurrentSlide - 1 > 0)
        {
            <button type="button" class="btn btn-primary" @onclick="NavigateBackwards">Go Back</button>
        }
        @if (CurrentSlide + 1 <= NumberOfItems)
        {
            <button type="button" class="btn btn-primary" @onclick="NavigateForwards">Next Step</button>
        }
    </div>
</DynamicElement>
```

Components/BSCarouselCustomControlsBase.razor.cs
```using BlazorComponentUtilities;
using BlazorStrap;
using Microsoft.AspNetCore.Components;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace MyProject.Components
{
    public abstract class BSCarouselCustomControlsBase : ComponentBase
    {
        [Parameter(CaptureUnmatchedValues = true)] public IDictionary<string, object> UnknownParameters { get; set; }

        protected string Classname =>
         new CssBuilder("carousel-controls row justify-content-center pb-5")
             .AddClass(Class)
         .Build();

        [Parameter] public int NumberOfItems { get; set; }
        [Parameter] public string Class { get; set; }
        [CascadingParameter] internal BSCarousel Parent { get; set; }

        [Parameter] public EventCallback<int> ActiveIndexChanged { get; set; }

        public int CurrentSlide => Parent.ActiveIndex + 1;

        protected async Task NavigateForwards()
        {
            if (Parent.AnimationRunning || Parent.ActiveIndex + 1 > Parent.NumberOfItems - 1) return;
            Parent.ResetTimer();
            Parent.ActiveIndex++;
            await ActiveIndexChanged.InvokeAsync(Parent.ActiveIndex).ConfigureAwait(false);
        }

        protected async Task NavigateBackwards()
        {
            if (Parent.AnimationRunning || Parent.ActiveIndex - 1 < 0) return;
            Parent.ResetTimer();
            Parent.ActiveIndex--;
            await ActiveIndexChanged.InvokeAsync(Parent.ActiveIndex).ConfigureAwait(false);
        }
    }
}
```

To use, simply add `<BSCarouselCustomControls NumberOfItems="3" />` as the child of your `<BSCarousel>`

**Comment by MasterSkriptor** [comment]
@mark-at-tusksoft I'll take a look at this and get back to you with what I find out.

**Comment by MasterSkriptor** [comment]
@mark-at-tusksoft All of the examples on the Carousel pages appear to work fine for me in both Chromium Edge, and Chrome.

One thing I did notice is the typo you were referring to: `Clickable image example code line 4 has an LShft operator @for (int i = 0; i <<> items.Count; i++) likely a typo?`

This typo is the result of a bug in PrettyCode which we are using to display the Razor Code. I posted an issue in PrettyCode's repo to address this issue. The code should look like this:
**@for (int i = 0; i < items.Count; i++)**

Another question I wanted to ask is whether or not you hace tried the BSDataCarousel to see if it works better for what you are trying to accomplish?

I have added the IndexChanged method to the code example of the **With Captions & Headers & Fade** section of the help file.
